manifests/machineconfig.crd.yaml

```diff
@@ -39,205 +41,401 @@ spec:
+                  type: object
+                  properties:
+                    units:
+                      type: array
+                      items:
+                        type: object
+                        properties:
+                          contents:
+                            type: string
+                          dropins:
+                            type: array
+                            items:
+                              type: object
+                              properties:
+                                contents:
+                                  type: string
+                                name:
+                                  type: string
+                          name:
+                            type: string
                 passwd:
                   type: object
                   properties:
+                    groups:
+                      type: array
+                      items:
+                        type: object
+                        properties:
+                          gid:
+                            type: integer
+                          name:
+                            type: string
+                          passwordHash:
+                            type: string
+                          system:
+                            type: boolean
                     users:
                       type: array
                       items:
                         type: object
                         properties:
+                          create:
+                            type: object
+                            properties:
+                              gecos:
+                                type: string
+                              groups:
+                                type: array
+                                items:
+                                  type: string
+                              homeDir:
+                                type: string
+                              noCreateHome:
+                                type: boolean
+                              noLogInit:
+                                type: boolean
+                              noUserGroup:
+                                type: boolean
+                              primaryGroup:
+                                type: string
+                              shell:
+                                type: string
+                              system:
+                                type: boolean
+                              uid:
+                                type: integer
+                          gecos:
+                            type: string
+                          groups:
+                            type: array
+                            items:
+                              type: string
+                          homeDir:
+                            type: string
                           name:
-                            description: "name of user. Must be \"core\" user."
```

Comment:
How can we get these descriptions added back again? Would we need to add proper godoc comments to the types in the ignition repo? For ones like this where we further restrict what ignition we accept, how would be indicate that to users?

Reply:
Hm, good question. This is obviously a result of the auto generation, and in our case we probably want to make manual fixes for `machineconfig` that differs from general ignition usage. I don't think we should modify ignition directly, but keeping manual modifications may be costly.